reposerver/repository/repository.go

```diff
@@ -145,7 +145,7 @@ func (s *Service) runRepoOperation(
 	source *v1alpha1.ApplicationSource,
 	verifyCommit bool,
 	getCached func(revision string, firstInvocation bool) (bool, interface{}, error),
-	operation func(appPath, repoRoot, revision, verifyResult string) (interface{}, error),
+	operation func(appPath, repoRoot, commitSHA, revision, verifyResult string) (interface{}, error),
```

Comment:
We might want to add some commment to describe the difference between these parameters.   `commitSHA` is a resolved SHA from a revision.  It goes to the generated manifest.  `revision` is used as a key to manifest cache.  I would probably rename `revision` to `cacheKey`.